server/src/main/java/org/opensearch/search/pipeline/SearchPhaseInjectorProcessor.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.pipeline;
+
+import org.opensearch.action.search.SearchPhase;
+import org.opensearch.action.search.SearchPhaseContext;
+import org.opensearch.action.search.SearchPhaseResults;
+import org.opensearch.search.SearchPhaseResult;
+
+/**
+ * Creates a processor that runs between Phases of the Search.
+ */
+public interface SearchPhaseInjectorProcessor extends Processor {
```

Comment:
The idea of the processor is it runs between the phases. If we say it runs on phaseInput then we need to define on what phase input it should work, but here we want to run the processor when a certain phase is completed and a certain phase is next phase. 

Also response of a phase is always input to next phase. So give that understanding I don't think so we will ever have this phaseinput and output processor.

Reply:
> The idea of the processor is it runs between the phases. If we say it runs on phaseInput then we need to define on what phase input it should work,

The input is already defined here which is `SearchPhaseResults` and that is part of the interface hence the recommendation to make it more clear.

Reply:
It works on SearchPhaseResults but  on which searchphase results and when? That is not defined. The problem here is, the processor works on phase results but it also runs between phases. So i took a choice to call it injector processor. But if you look at SearchRequestProcessor because request automatically means that when it will be run. So no ambiguity. 
But with phase processors thats an ambiguity. 

This is the whole thought process behind this name. Here importance is given to where the processor is running rather than on what thing it is running.

Reply:
`When` is defined by `before` and `after` phase methods of this interface. IMO **what** it is processing should be reflected in name to be more intuitive not when. `Injector` keyword seems counter intuitive to me.

Reply:
This seems to be valid point. I think I am on borderline from flipping the name. Before I do that I would like to know if anyone else has any other point to add. @msfroh @nknize @dblock let us know your thoughts. If we align on SearchPhaseResultsProcessor I will make the change.